Add `@` shorthand for `CoordinateSystem` methods `coords_to_point` (`c2p`) and `point_to_coords` (`p2c`)

JasonGrace2282 opened the pull request:
Allows `ax @ (1, 0, 0)` for `coords_to_point` and `(1, 0, 0) @ ax` for `point_to_coords`

Inspired by @abul4fia

behackl reviewed (approved):
I like this a lot! 👍 

It might make sense modify or include a new example in the example gallery that uses this notation.

JasonGrace2282 commented on manim/mobject/graphing/coordinate_systems.py:
```diff
@@ -1779,6 +1780,14 @@ def construct(self):
+    def __matmul__(self, coord: Point3D | Mobject):
+        if isinstance(coord, Mobject):
+            coord = coord.get_center()
+        return self.coords_to_point(*coord)
```
The problem with removing the `*` is that `ax @ (1, 0, 0)` then returns a 3x3 array, which I felt was kinda unintuitive.

JasonGrace2282 commented:
I just added it to `NumberLine` and changed the `CoordinateSystem` example to use `@` notation. In the future, maybe we should transition to that being the "standard" (although `c2p` can always exist for more complex expressions)

Viicos commented:
Will get back to the unresolved comment soon

Viicos commented on manim/mobject/graphing/coordinate_systems.py:
```diff
@@ -144,7 +145,7 @@ def __init__(
         self.num_sampled_graph_points_per_tick = 10
 
-    def coords_to_point(self, *coords: Sequence[ManimFloat]):
+    def coords_to_point(self, *coords: Point3D):
```
```suggestion
    def coords_to_point(self, *coords: ManimFloat):
```

This is what got me confused regarding my other comment
